### ISSUE TITLE
crash with std::clamp

### ISSUE THREAD

**Issue by michalmoc** [variar/klogg#592]
klogg crashes with 

```
/usr/include/c++/13.1.1/bits/stl_algo.h:3669: constexpr const _Tp& std::clamp(const _Tp&, const _Tp&, const _Tp&) [with _Tp = int]: Assertion '!(__hi < __lo)' failed.
fish: Job 1, 'klogg' terminated by signal SIGABRT (Przerwij)
```

To reproduce issue it's enough to perform 'drag and drop' gesture inside the log area, which is easy enough to do accidentaly. Since there seem to be no autosaves, such crash means loss of work, which renders Klogg unusable.

Arch linux, build from AUR, CMake output:

```
-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Version override 22.06.0.1289
Build 
Project version is 22.06.0
Project version tweak is 1289
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1") 
Flags set for 'GNU' compiler:-ffast-math
Defines set for 'GNU' compiler:

CMAKE_C_FLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -mmmx -msse -msse2 -msse3 -mssse3 -mpopcnt -march=native -mtune=generic
CMAKE_CXX_FLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -fno-sized-deallocation -fno-omit-frame-pointer -mmmx -msse -msse2 -msse3 -mssse3 -mpopcnt -march=native -mtune=generic
CMAKE_EXE_LINKER_FLAGS: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
CMAKE_MODULE_LINKER_FLAGS: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
CMAKE_SHARED_LINKER_FLAGS: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
CMAKE_STATIC_LINKER_FLAGS: 
CMAKE_C_FLAGS_RELWITHDEBUGINFO: 
CMAKE_CXX_FLAGS_RELWITHDEBUGINFO: 
CMAKE_EXE_LINKER_FLAGS_RELWITHDEBUGINFO: 
CMAKE_MODULE_LINKER_FLAGS_RELWITHDEBUGINFO: 
CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBUGINFO: 
CMAKE_STATIC_LINKER_FLAGS_RELWITHDEBUGINFO: 

Qt version 5.15.9
LTO ON
-- Found Git: /usr/bin/git (found version "2.40.1") 
-- Found Python3: /usr/bin/python3.11 (found version "3.11.3") found components: Interpreter 
-- Found the following ICU libraries:
--   uc (required): /usr/lib/libicuuc.so
-- Found ICU: /usr/include (found version "72.1") 
-- CPM: adding package simdutf@1.0.1 (v1.0.1 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/simdutf/941e9766f01c5505fb4646c29406412bcfc0d924)
CMake Deprecation Warning at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/croaring/fb2628131c81ddf8e7b8973f7a0db62a4aa5e082/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/croaring/fb2628131c81ddf8e7b8973f7a0db62a4aa5e082/CMakeLists.txt:2 (project):
  Policy CMP0048 is not set: project() command manages VERSION variables.
  Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The following variable(s) would be set to empty:

    PROJECT_VERSION
    PROJECT_VERSION_MAJOR
    PROJECT_VERSION_MINOR
    PROJECT_VERSION_PATCH
    PROJECT_VERSION_TWEAK
This warning is for project developers.  Use -Wno-dev to suppress it.

-- BENCHMARK_DATA_DIR: /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/croaring/fb2628131c81ddf8e7b8973f7a0db62a4aa5e082/benchmarks/realdata/
-- TEST_DATA_DIR: /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/croaring/fb2628131c81ddf8e7b8973f7a0db62a4aa5e082/tests/testdata/
-- Building a static library.
-- ROARING_LIB_TYPE: STATIC
-- Library output directory (does not apply to Visual Studio): /home/mmoczuls/.cache/yay/klogg/src/klogg-22.06/build
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- CMAKE_BUILD_TYPE: RelWithDebugInfo
-- ROARING_DISABLE_NEON: OFF
-- ROARING_BUILD_STATIC: ON
-- ROARING_LINK_STATIC: OFF
-- ROARING_BUILD_LTO: OFF
-- ROARING_SANITIZE: OFF
-- CMAKE_C_COMPILER: /usr/bin/cc
-- CMAKE_C_FLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -mmmx -msse -msse2 -msse3 -mssse3 -mpopcnt -march=native -mtune=generic -std=c11 -fPIC   -Wall -Wextra -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self  
-- CMAKE_C_FLAGS_DEBUG: -ggdb
-- CMAKE_C_FLAGS_RELEASE: -O3
-- ROARING_EXCEPTIONS: ON
-- CPM: adding package CRoaring@0 (d582cd224ebc2f5a1585035a8f5ce306052e8ad1 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/croaring/fb2628131c81ddf8e7b8973f7a0db62a4aa5e082)
-- CPM: adding package maddy@0 (67d331c59d3abc4e92eecc5874836a600dbfaf58 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/maddy/bb6b5ce266bb630fda8e2296d64604959cd99c66)
-- CPM: using local package hyperscan@0
-- CPM: adding package NamedType@0 (215f039574a7d5111f1a131def337f142029f896 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/namedtype/178f0d7023d4ba2b25cfd0136a4eb140edf53ce2)
Adding imported target for NamedType
-- CPM: using local package Uchardet@0.0.7
-- CPM: using local package KF5Archive@0
-- CPM: adding package robin_hood@3.11.2 (3.11.2 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/robin_hood/93816bbe27b7663cb4573f014b8cf1ef4d81954b)
-- CPM: adding package SingleApplication@3.3.4 (v3.3.4 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/singleapplication/23aa3f1b96d47fedade6f45b8b73e103ad2a2fb8)
-- CPM: using local package xxHash@0.8.1
-- CPM: adding package whereami@0 (ba364cd54fd431c76c045393b6522b4bff547f50 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/whereami/3af7d0aa7cde59af067af5687d050c40b75e8864)
-- CPM: adding package exprtk@0 (1f9f4cd7d2620b7b24232de9ea22908d63913459 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/exprtk/17be605a2fb439d5361d50b80f449c51cd094951)
-- CPM: adding package KDToolBox@0 (253a6c087c626cd477e15c9dae4b0e3ec27afaee at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/kdtoolbox/05e252c60f3358beaddb2ef5e797dd406acef0e0)
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- CPM: adding package efsw@0 (2e73f8e96779a2895c6242d66cadb3a15aa5e192 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/efsw/8287556e3d7c75ff9984a7a4503e67404ab86906)
-- CPM: using local package tbb@0
-- Found tbbmalloc_proxy: /usr/lib/libtbbmalloc_proxy.so  
-- 
-- Override standard malloc (MI_OVERRIDE=ON)
-- Set full secure build (MI_SECURE=ON)
-- 
-- Library base name: mimalloc-secure-relwithdebuginfo
-- Version          : 2.0
-- Build type       : relwithdebuginfo
-- C Compiler       : /usr/bin/cc
-- Compiler flags   : -Wall;-Wextra;-Wno-unknown-pragmas;-fvisibility=hidden;-Wstrict-prototypes;-ftls-model=initial-exec;-fno-builtin-malloc
-- Compiler defines : MI_SECURE=4
-- Link libraries   : /usr/lib/libpthread.a;/usr/lib/librt.a;atomic
-- Build targets    : object
-- 
-- CPM: adding package mimalloc@2.0.5 (v2.0.5 at /home/mmoczuls/.cache/yay/klogg/src/cpm_cache/mimalloc/434fda146b5c5bc88251218d40a844db08314a7d)
-- Configuring done (1.9s)
-- Generating done (0.1s)
-- Build files have been written to: /home/mmoczuls/.cache/yay/klogg/src/klogg-22.06/build
```

**Comment by variar** [comment]
Thanks for the report. I checked usages of std::clamp in current code and added checks for boundaries.  Should be available in 23.06.0.1418+. klogg-git aur package should pick this version up when there is new github release.

**Comment by michalmoc** [comment]
Thanks for quick reaction. For now it seems to be fixed.
